From patchwork Sat Nov  7 21:47:56 2009
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [10/75] V4L/DVB: declare MODULE_FIRMWARE for modules using XC2028 and
	XC3028L tuners
Date: Sat, 07 Nov 2009 21:47:56 -0000
From: Ben Hutchings <ben@decadent.org.uk>
X-Patchwork-Id: 58394

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>

---
I'm not really sure whether it's better to do this in the drivers which
specify which firmware file to use, or just once in the xc2028 tuner
driver.  Your call.

Ben.

 drivers/media/dvb/dvb-usb/cxusb.c           |    1 +
 drivers/media/dvb/dvb-usb/dib0700_devices.c |    1 +
 drivers/media/video/cx18/cx18-driver.c      |    1 +
 drivers/media/video/cx23885/cx23885-dvb.c   |    3 +++
 drivers/media/video/cx88/cx88-cards.c       |    2 ++
 drivers/media/video/em28xx/em28xx-cards.c   |    3 +++
 drivers/media/video/ivtv/ivtv-driver.c      |    1 +
 drivers/media/video/saa7134/saa7134-cards.c |    2 ++
 8 files changed, 14 insertions(+), 0 deletions(-)

--- a/drivers/media/dvb/dvb-usb/cxusb.c
+++ b/drivers/media/dvb/dvb-usb/cxusb.c
@@ -1863,3 +1863,4 @@ MODULE_AUTHOR("Chris Pascoe <c.pascoe@it
 MODULE_DESCRIPTION("Driver for Conexant USB2.0 hybrid reference design");
 MODULE_VERSION("1.0-alpha");
 MODULE_LICENSE("GPL");
+MODULE_FIRMWARE(XC2028_DEFAULT_FIRMWARE);
--- a/drivers/media/dvb/dvb-usb/dib0700_devices.c
+++ b/drivers/media/dvb/dvb-usb/dib0700_devices.c
@@ -408,6 +408,7 @@ static struct xc2028_ctrl stk7700ph_xc30
 	.max_len = 64,
 	.demod = XC3028_FE_DIBCOM52,
 };
+MODULE_FIRMWARE(XC2028_DEFAULT_FIRMWARE);
 
 static struct xc2028_config stk7700ph_xc3028_config = {
 	.i2c_addr = 0x61,
--- a/drivers/media/video/cx18/cx18-driver.c
+++ b/drivers/media/video/cx18/cx18-driver.c
@@ -237,6 +237,7 @@ MODULE_AUTHOR("Hans Verkuil");
 MODULE_DESCRIPTION("CX23418 driver");
 MODULE_SUPPORTED_DEVICE("CX23418 MPEG2 encoder");
 MODULE_LICENSE("GPL");
+MODULE_FIRMWARE(XC2028_DEFAULT_FIRMWARE);
 
 MODULE_VERSION(CX18_VERSION);
 
--- a/drivers/media/video/cx23885/cx23885-dvb.c
+++ b/drivers/media/video/cx23885/cx23885-dvb.c
@@ -932,6 +932,9 @@ static int dvb_register(struct cx23885_t
 	return ret;
 }
 
+MODULE_FIRMWARE(XC2028_DEFAULT_FIRMWARE);
+MODULE_FIRMWARE(XC3028L_DEFAULT_FIRMWARE);
+
 int cx23885_dvb_register(struct cx23885_tsport *port)
 {
 
--- a/drivers/media/video/cx88/cx88-cards.c
+++ b/drivers/media/video/cx88/cx88-cards.c
@@ -3080,6 +3080,8 @@ void cx88_setup_xc3028(struct cx88_core 
 }
 EXPORT_SYMBOL_GPL(cx88_setup_xc3028);
 
+MODULE_FIRMWARE(XC2028_DEFAULT_FIRMWARE);
+
 static void cx88_card_setup(struct cx88_core *core)
 {
 	static u8 eeprom[256];
--- a/drivers/media/video/em28xx/em28xx-cards.c
+++ b/drivers/media/video/em28xx/em28xx-cards.c
@@ -2083,6 +2083,9 @@ static void em28xx_setup_xc3028(struct e
 	}
 }
 
+MODULE_FIRMWARE(XC2028_DEFAULT_FIRMWARE);
+MODULE_FIRMWARE(XC3028L_DEFAULT_FIRMWARE);
+
 static void em28xx_tuner_setup(struct em28xx *dev)
 {
 	struct tuner_setup           tun_setup;
--- a/drivers/media/video/ivtv/ivtv-driver.c
+++ b/drivers/media/video/ivtv/ivtv-driver.c
@@ -254,6 +254,7 @@ MODULE_SUPPORTED_DEVICE
     ("CX23415/CX23416 MPEG2 encoder (WinTV PVR-150/250/350/500,\n"
 		"\t\t\tYuan MPG series and similar)");
 MODULE_LICENSE("GPL");
+MODULE_FIRMWARE(XC2028_DEFAULT_FIRMWARE);
 
 MODULE_VERSION(IVTV_VERSION);
 
--- a/drivers/media/video/saa7134/saa7134-cards.c
+++ b/drivers/media/video/saa7134/saa7134-cards.c
@@ -6975,6 +6975,8 @@ static void saa7134_tuner_setup(struct s
 	}
 }
 
+MODULE_FIRMWARE(XC2028_DEFAULT_FIRMWARE);
+
 /* stuff which needs working i2c */
 int saa7134_board_init2(struct saa7134_dev *dev)
 {
